packages/rs-dpp/src/data_contract/document_type/class_methods/try_from_schema/v0/mod.rs

```diff
@@ -426,20 +479,13 @@ fn insert_values(
 
         let mut inner_properties = property_value.to_btree_ref_string_map()?;
 
-        if let Some(schema_ref) = inner_properties
-            .get_optional_str(property_names::REF)
-            .map_err(ProtocolError::ValueError)?
-        {
-            let referenced_sub_schema = resolve_uri(root_schema, schema_ref).map_err(|err| {
-                ProtocolError::Generic(format!("invalid schema reference url: {err}"))
-            })?;
+        if let Some(schema_ref) = inner_properties.get_optional_str(property_names::REF)? {
+            let referenced_sub_schema = resolve_uri(root_schema, schema_ref)?;
```

Comment:
This will through an error if ref is invalid. I think it's a user error

Reply:
Actually no, because it gets converted to a DataContractError (Which is the result). And DataContractErrors can be consensus errors.

Reply:
Oh, ok

Reply:
They get transformed into consensus errors with
`#[inline]
fn consensus_or_protocol_data_contract_error(
    data_contract_error: DataContractError,
) -> ProtocolError {
    #[cfg(feature = "validation")]
    {
        ProtocolError::ConsensusError(
            ConsensusError::BasicError(BasicError::ContractError(data_contract_error)).into(),
        )
    }
    #[cfg(not(feature = "validation"))]
    {
        ProtocolError::DataContractError(data_contract_error)
    }
}
`

The validation feature is needed for the abci app, hence they will be consensus errors there. On clients however we will get protocol errors.

Reply:
Few calls weren't covered with map_err